src/types.ts

```diff
@@ -1,18 +1,64 @@
+import { asMap, asNumber, asObject, asOptional, asString } from 'cleaners'
 import * as Nano from 'nano'
 
-export interface StoreRoot extends Nano.MaybeDocument {
-  files: StoreFileMap
+// Is there a better way to make optional properties for this type?
+const asNanoMaybeDocument = asObject<Nano.MaybeDocument>({
+  _id: asOptional(asString),
+  _rev: asOptional(asString)
+})
+
+export type StoreFileMap = ReturnType<typeof asStoreFileMap>
+export const asStoreFileMap = asMap(asNumber)
+
+// Directory
+
+export type StoreDirectory = ReturnType<typeof asStoreDirectory>
+export type StoreDirectoryDocument = ReturnType<typeof asStoreDirectoryDocument>
+export const asStoreDirectory = asObject({
+  deleted: asStoreFileMap,
+  paths: asStoreFileMap
+})
+export const asStoreDirectoryDocument = asObject({
+  ...asNanoMaybeDocument.shape,
+  ...asStoreDirectory.shape
+})
+
+// Root
+
+export interface StoreRoot extends StoreDirectory {
```

Comment:
duplicate from below

Reply:
The reason for the duplicate static type is for the optional properties `lastGitHash` and `lastGitTime`. The inferred type from the cleaner does not allow for these properties to be optional. The same thing is done for the `NanoMaybeDocument` type.